saleor/graphql/attribute/utils.py

```diff
@@ -137,21 +137,59 @@ def _pre_save_values(
     ):
         """Lazy-retrieve or create the database objects from the supplied raw values."""
         get_or_create = attribute.values.get_or_create
-        is_numeric_attr = attribute.input_type == AttributeInputType.NUMERIC
-
-        def get_slug_value(value):
-            slug_value = value if not is_numeric_attr else value.replace(".", "_")
-            return slugify(slug_value, allow_unicode=True)
 
         return tuple(
             get_or_create(
                 attribute=attribute,
-                slug=get_slug_value(value),
+                slug=slugify(value, allow_unicode=True),
                 defaults={"name": value},
             )[0]
             for value in attr_values.values
         )
 
+    @classmethod
+    def _pre_save_numeric_values(
+        cls,
+        instance: T_INSTANCE,
+        attribute: attribute_models.Attribute,
+        attr_values: AttrValuesInput,
+    ):
+        defaults = {
+            "name": attr_values.values[0],
+        }
+        return cls._update_or_create_value(instance, attribute, defaults)
+
+    @classmethod
+    def _pre_save_rich_text_values(
+        cls,
+        instance: T_INSTANCE,
+        attribute: attribute_models.Attribute,
+        attr_values: AttrValuesInput,
+    ):
+        defaults = {
+            "rich_text": attr_values.rich_text,
+            "name": truncatechars(
+                clean_editor_js(attr_values.rich_text, to_string=True), 50
+            ),
+        }
+        return cls._update_or_create_value(instance, attribute, defaults)
+
+    @classmethod
+    def _update_or_create_value(
+        cls,
+        instance: T_INSTANCE,
+        attribute: attribute_models.Attribute,
+        value_defaults: dict,
+    ):
+        update_or_create = attribute.values.update_or_create
+        slug = slugify(f"{instance.id}_{attribute.id}", allow_unicode=True)
+        value = update_or_create(
+            attribute=attribute,
+            slug=slug,
+            defaults=value_defaults,
+        )[0]
+        return (value,)
```

Comment:
maybe more generic django-like signature, like this functions with `_or_create` returns (created, value) ?